### PR TITLE
Vp9

### PR DESCRIPTION
**Pull request by feiwan1**

### CODE REVIEW

File: libavcodec/decode.c [hunk 1]
```diff
@@ -1202,6 +1229,15 @@ int ff_get_format(AVCodecContext *avctx, const enum AVPixelFormat *fmt)
         return AV_PIX_FMT_NONE;
 
     for (;;) {
+        if (avctx->internal->hwaccel_priv_data &&
+            avctx->hwaccel->caps_internal & HWACCEL_CAP_INTERNAL_ALLOC) {
+            err = hwaccel_realloc_surface(avctx);
```
Comment (xhaihao) [inline libavcodec/decode.c:1234]: Is a flush needed before realllocating surface ?

Comment (feiwan1) [reply libavcodec/decode.c:1234]: > Is a flush needed before realllocating surface ?

The previous surface and hw_frames_ctx is stored/referenced by each AVFrame. So just need to create new hw_frames_ctx, and the new AVFrame will pick up surface from it. The old AVFrame still keep there with old surface if it is put into reference list. So don't need to flush it.

File: libavcodec/decode.c [hunk 2]
```diff
@@ -1156,6 +1156,26 @@ static void hwaccel_uninit(AVCodecContext *avctx)
...
+        for (i = 0;; i++) {
+            hw_config = ffcodec(avctx->codec)->hw_configs[i];
+            if (!hw_config)
+                break;
+            if (hw_config->public.pix_fmt == fmt)
+                break;
+        }
+    } else {
+        hw_config = NULL;
+    }
```
Comment (xhaihao) [inline libavcodec/decode.c:1174]: the else statement is not needed if hw_config is set to NULL when declaring it.

Comment (xhaihao) [reply libavcodec/decode.c:1174]: > the else statement is not needed if hw_config is set to NULL when declaring it.

Please ignore this comment.

**Review by xhaihao — APPROVED** [review]
LGTM

**Comment by feiwan1** [comment]
https://patchwork.ffmpeg.org/project/ffmpeg/list/?series=7206

**Pull request closed without merging**